src/types/bytearray.rs

```diff
@@ -264,22 +283,110 @@ mod test {
     }
 
     #[test]
-    fn test_byte_array_new_with() {
+    fn test_byte_array_new_with() -> super::PyResult<()> {
         let gil = Python::acquire_gil();
         let py = gil.python();
         let py_bytearray = PyByteArray::new_with(py, 10, |b: &mut [u8]| {
             b.copy_from_slice(b"Hello Rust");
-        });
+            Ok(None)
+        })?;
         let bytearray: &[u8] = unsafe { py_bytearray.as_bytes() };
         assert_eq!(bytearray, b"Hello Rust");
+        Ok(())
     }
 
     #[test]
-    fn test_byte_array_new_with_zero_initialised() {
+    fn test_byte_array_new_with_zero_initialised() -> super::PyResult<()> {
         let gil = Python::acquire_gil();
         let py = gil.python();
-        let py_bytearray = PyByteArray::new_with(py, 10, |_b: &mut [u8]| ());
+        let py_bytearray = PyByteArray::new_with(py, 10, |_b: &mut [u8]| Ok(None))?;
         let bytearray: &[u8] = unsafe { py_bytearray.as_bytes() };
         assert_eq!(bytearray, &[0; 10]);
+        Ok(())
+    }
+
+    #[test]
+    fn test_byte_array_new_with_truncated() -> super::PyResult<()> {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let py_bytearray = PyByteArray::new_with(py, 10, |b: &mut [u8]| {
+            b.copy_from_slice(b"Hello Rust");
+            Ok(Some(5))
+        })?;
+        let bytearray: &[u8] = unsafe { py_bytearray.as_bytes() };
+        assert_eq!(bytearray, b"Hello");
+        Ok(())
+    }
+
+    #[test]
+    fn test_byte_array_new_with_no_growing() -> super::PyResult<()> {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let py_bytearray = PyByteArray::new_with(py, 10, |b: &mut [u8]| {
+            b.copy_from_slice(b"Hello Rust");
+            Ok(Some(15))
+        })?;
+        let bytearray: &[u8] = unsafe { py_bytearray.as_bytes() };
+        assert_eq!(bytearray, b"Hello Rust");
+        Ok(())
+    }
+
+    #[test]
+    fn test_byte_array_new_with_error() -> super::PyResult<()> {
+        use crate::exceptions::PyValueError;
+        use crate::types::PyDict;
+
+        let alloc_size = 1024 * 1024;
+
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let locals = PyDict::new(py);
+
+        py.run(
```

Comment:
I don't like this kind of test since it can be **really** unstable because of GC.
We can confirm that `DECREF` is called and it's sufficient. I think it's too nervous to test the memory difference.

Reply:
If just the comment is ok, I'll gladly remove that test :)

Reply:
This test has been removed in 4b3422e3b43d5fc43f430648f6479f1e2c4fcf18